### ISSUE TITLE
[suggestion] Use vanilla container as stock block

### ISSUE THREAD

**Issue by Uboaaaaa** [Beez0r/iShop#3]
How about integrate the stock function into vanilla containers?
So the stock would be placed directly into a specific container defined by the player, where the good is extracted.
This could make the experience a bit more immersive imo. One cannot walk to a trash can someone placed by the road, click it and somehow refill their shop's stock through it.

**Comment by Beez0r** [comment]
There is a config option **stockBlock** to change the stock block to any Minecraft block you want. Specified stock block will only open the stock GUI if left clicked; so chests, ender chests, barrels, composter, or anything with a right click use can still be functional and used along with being a stock block as well. The only change when setting stock block the same as a vanilla container or usable block is **players must be sneaking (holding shift key) to destroy the stock block**

You could also disable stock blocks all together by setting config option **enableStockBlock** to false, **enableStockCommand** to true, and only use the **/shop stock** command.

**Comment by Beez0r** [comment]
There is a config option **stockBlock** to change the stock block to any Minecraft block you want. Specified stock block will only open the stock GUI if left clicked; so chests, ender chests, composter, or anything with a right click use can still be functional and used along with being a stock block as well. The only change when setting stock block the same as a vanilla container or usable block is **players must be sneaking (holding shift key) to destroy the stock block**

You could also disable stock blocks all together by setting config option **enableStockBlock** to false, **enableStockCommand** to true, and only use the /shop stock command.

**Comment by Uboaaaaa** [comment]
Sorry if that was confusing, I was talking about a separate feature in my mind.

Currently the **stock** exists as an independent inventory that can be accessed everywhere, either by using command or clicking blocks of a certain type.

I think it'll be interesting if the stock block functions _locally_. For example, one shop is tied to a specific stockBlock around it which has its own content. There is also a maximum distance where the stock block has to be form the shop block so that it can be refilled successfully.

With that idea in mind I think maybe it can use vanilla minecraft chest inventories as the described _local stockBlock_ where items are taken from. A player has to link a chest to a shop with command to set the chest as the stock. The chest is then protected from others modification and items are taken if a purchase is made.

**Comment by Beez0r** [comment]
This is a great feature request that has been closely requested before!

I will soon begin development on having shared shops linked to shared stock between players. Using your suggestion, options of requiring a max distance between shop and stock block locations could be implemented. This will take some time and testing to perfect overall, but is possible.

I must also preserve and respect options for those who do not want to use new features.

**Comment by Beez0r** [comment]
I am still slowly working on this! I have been more focused on adding other new features and fixing bugs in the current latest versions before fully committing to this big feature release

**Comment by Beez0r** [comment]
I've implemented a new feature in the latest update 2.13 that will set a distance requirement between shops and stock if enabled in the config, this helps solve the issue of being able to access stock from anywhere. So far I haven't been able to implement specific stock tied to specific shops without completely reworking the database, wiping all data, and some laggy workarounds which is not practical. Another feature that was added a day after this issue was opened is **enableStockAccessFromShopGUI**, which can be useful for this case by eliminating stock blocks all together. Still seeing what I can do...
